### PR TITLE
zd14249

### PR DESCRIPTION
**Pull request by SKlimaRA**
# Description

Added build flag to make DoVerifyCallback verify hostname or ip addr only against altnames instead of altnames and common name.

Fixes zd#14249

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
OK to test. @SKlimaRA is an approved contributor.

File: src/internal.c [hunk 1]
```diff
@@ -10500,6 +10500,7 @@ int CheckHostName(DecodedCert* dCert, const char *domainName, size_t domainNameL
         ret = 0;
     }
 
+#ifndef ONLY_ALT_NAME_VERIFICATION
```
Comment (dgarske) [inline src/internal.c:10503]: Can we rename this to something like `WOLFSSL_HOSTNAME_VERIFY_ALT_NAME_ONLY`? Please add description of build option at top of internal.c with others.

File: src/internal.c [hunk 2]
```diff
@@ -10509,6 +10510,7 @@ int CheckHostName(DecodedCert* dCert, const char *domainName, size_t domainNameL
             WOLFSSL_MSG("DomainName match on common name failed");
         }
     }
+#endif /* #ifndef ONLY_ALT_NAME_VERIFICATION */
```
Comment (dgarske) [inline src/internal.c:10513]: Syntax should be `#endif /* !ONLY_ALT_NAME_VERIFICATION */`

**Review by dgarske — CHANGES_REQUESTED** [review]
Please share a test case for reproducing. Thanks

**Comment by SKlimaRA** [comment]
I will share the test case later. Please wait. Thanks.

**Comment by SKlimaRA** [comment]
Hey @dgarske ,
could you please desribe the test case you want.
I'm using our server and client to testing this and I can't share the code for these. But I could share the certificates and keys for this test case.
Thanks.

**Review by dgarske — CHANGES_REQUESTED** [review]
Is it possible to refactor the code to use the common name if subject alternate name (SAN) is not provided in the certificate? Or is it your intention to reject certificates without a SAN with this build option.

Test case is tough since it is a build option and we'll need to add to our internal CI. I'll have to think about it. Perhaps let's discuss in the ZenDesk ticket?

**Review by dgarske — APPROVED** [review]
Thanks @SKlimaRA . I had missed your latest commit. I'd like another engineer @SparkiDev to also review this change before it gets merged.

**Comment by SparkiDev** [comment]
ProcessPeerCerts() checks domain name.
Did you consider changes to this function as well?

**Comment by SKlimaRA** [comment]
Hi @SparkiDev 
I guess we could replicate the changes to ProcessPeerCerts() but I don't think it's necessary for us now.
Please let me know if you would prefer making the changes to ProcessPeerCerts().
Thanks

**Comment by SparkiDev** [comment]
This PR fixes the ticket, so I'll merge this as is.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**